packages/perspective-viewer-d3fc/src/js/legend/styling/draggableComponent.js

```diff
@@ -39,12 +40,13 @@ export function draggableComponent() {
             };
             settings.legend = {...settings.legend, ...position};
 
-            if (isNodeInTopRight(node)) {
-                pinned = pinNodeToTopRight(node);
-                return;
-            }
+            pinned = isNodeInTopRight(node)
+                ? pinNodeToTopRight(node)
+                : unpinNodeFromTopRight(node, pinned);
+        });
 
-            pinned = unpinNodeFromTopRight(node, pinned);
+        drag.on("end", function (event) {
```

Comment:
👍 Thanks for making this update. It did cross my mind that we probably want to fire the event after the user stops updating the position, but wasn't 100% sure how to do that here.